### DISCUSSION TITLE
suggestions for the table component and the text component

### DISCUSSION THREAD

**Discussion by SebastiendOrnano** [sqlpage/SQLPage#740] [Ideas]
### Table component
Could it be possible to introduce a title parameter as in the datagrid and list component ?

### Text component
Could it be possible to place the id parameter inside the div element like in the datagrid component and not before the div ?

in the datagrid handlebar 
`data grid <div class="card my-2 {{class}}" {{#if id}}id="{{id}}"{{/if}}>`

in the present version of the text handlebar
`{{{~html~}}}
{{~#if title~}}
    <h1 id="{{id}}" class="mt-3 {{#if center}}text-center{{/if}}">{{title}}</h1>
{{~else~}}
    {{#if id}}
        <a id="{{id}}"></a>
    {{/if}}
{{~/if~}}
{{~#if contents_md~}}
    <div class="remove-bottom-margin">{{{markdown contents_md}}}</div>`

if the handlebar could be sliglty changed to 

'<div class="remove-bottom-margin"   {{#if id}}id="{{id}}"{{/if}}>{{{markdown contents_md}}}</div>`

the id parameter will be include in the div
In the present code, the id can be used as an anchor but can't be used to apply a specific style.
It could be nice to color the background or the border of a specific paragraph !

Anyway, thanks for the work you have already done, sqlpage is a fine solution for no pro like me !

**Comment by lovasoa** [comment]
Hello ! Both ideas sound good. Since it sounds like you already have a clear idea of the changes to make, can you open pull requests ? Don't forget to update the documentation too !

In the meantime, you can always use css sibling selectors to style the elements you need: https://developer.mozilla.org/en-US/docs/Web/CSS/Next-sibling_combinator

**Reply by SebastiendOrnano** [reply]
I discover the next-sibling combinator. It gives exactly the rendering I was looking for.

 _(but still,  it would be simpler to have the id embedded as in datagrid  ! )_

Thanks
